scripts/jenkins/ses/ansible/roles/ses/tasks/salt_setup.yml

```diff
@@ -41,10 +41,24 @@
 
 - meta: flush_handlers
 
-- name: "Wait for salt-minion to initialize"
-  pause:
-    seconds: 5
+- name: Ensure salt-minion is initialized
+  command: "salt-key -L"
+  register: minion
+  changed_when: false
+  failed_when: ansible_hostname not in minion.stdout_lines
+  until: minion is succeeded
+  retries: 10
+  delay: 1
 
 - name: Accept salt keys
   command: "salt-key --accept-all -y"
   changed_when: false
+
+- name: Ensure salt-minion is up
+  command: "salt-run manage.up"
+  register: minion_up
+  changed_when: false
+  failed_when: "'- ' ~ ansible_hostname not in minion_up.stdout_lines"
```

Comment:
This looks very fragile, shouldn't we just do something with a regex? Or will this stay stable?

Reply:
should be better now that I am at least enforcing the output format, besides using regex. If you have better idea please share ;)